### PR TITLE
feat: add searchEngine dropdown into searchinput

### PR DESCRIPTION
**Pull request by chiqors**
This is my first PR to implement small feature. Apologize, if there's some conventional that i should have used here.

Commits
✨ feat: add searchengine dropdown into searchinput component
🚧 wip: it's still using default html selectbox

### CODE REVIEW

**Comment by chiqors** [comment]
This PR for Issue #12

File: src/ui/components/search/SearchInput.tsx [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
+import { FaSpotify, FaYoutube } from "react-icons/fa";
 
 import { doSearch } from "@backend/search";
 import emitter from "@backend/events";
+import { getFromPath, saveFromPath } from "@backend/settings";
 import { router } from "@app/main";
 import { contentRoutes } from "@app/constants";
 
 import "@css/layout/TopBar.scss";
+import { SearchEngine } from "@app/backend/types";
```
Comment (KingRainbow44) [inline src/ui/components/search/SearchInput.tsx]: correct imports to match existing format (including ordering by import length)

File: src/ui/components/search/SearchInput.tsx [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
+interface IState {
+    searchType: SearchEngine;
+}
 
 class SearchInput extends React.Component<IProps, IState> {
     searchTimeout: NodeJS.Timeout | number = null;
 
     constructor(props: IProps) {
         super(props);
+        this.state = {
```
Comment (KingRainbow44) [inline src/ui/components/search/SearchInput.tsx]: apply spacing

File: src/ui/components/search/SearchInput.tsx [hunk 3]
```diff
@@ -64,6 +78,24 @@ class SearchInput extends React.Component<IProps, IState> {
...
+                    </option>
+                    <option value={"YouTube"}>
+                        <FaYoutube />
+                        YouTube
+                    </option>
+                    <option value={"Spotify"}>
+                        <FaSpotify />
+                        Spotify
+                    </option>
+                </select>
```
Comment (KingRainbow44) [inline src/ui/components/search/SearchInput.tsx]: apply spacing

**Review by KingRainbow44 — CHANGES_REQUESTED** [review]

**Comment by chiqors** [comment]
would this be enough?

File: src/ui/components/search/SearchInput.tsx [hunk 4]
```diff
@@ -7,12 +7,16 @@ import { FaSpotify, FaYoutube } from "react-icons/fa";
...
 import { contentRoutes } from "@app/constants";
+import { router } from "@app/main";
 
-import "@css/layout/TopBar.scss";
 import { SearchEngine } from "@app/backend/types";
 
+import "@css/layout/TopBar.scss";
+
+// -------------------------------------------------
+
```
Comment (KingRainbow44) [inline src/ui/components/search/SearchInput.tsx]: Remove unnecessary lines.

**Review by KingRainbow44 — CHANGES_REQUESTED** [review]

**Comment by KingRainbow44** [comment]
havent gotten around to testing this, will soon i promise lol

**Comment by KingRainbow44** [comment]
this was manually merged & updated in `080990ac30135d077de0fd8aa29a24b16a089507`

**Pull request closed without merging**